src/ol_infrastructure/applications/concourse/__main__.py

```diff
@@ -0,0 +1,532 @@
+    address=concourse_db.db_instance.address,
+    datacenter=consul_datacenter,
+)
+
+concourse_db_consul_service = Service(
+    "concourse-instance-db-service",
+    node=concourse_db_consul_node.name,
+    name="concourse-postgres",
+    port=concourse_db_config.port,
+    meta={
+        "external-node": True,
+        "external-probe": True,
+    },
+    checks=[
+        ServiceCheckArgs(
+            check_id="concourse-instance-db",
+            interval="10s",
+            name="concourse-instance-id",
+            timeout="60s",
+            status="passing",
+            tcp=f"{concourse_db.db_instance.address}:{concourse_db_config.port}",  # noqa: WPS237,E501
+        )
+    ],
+)
+
+##########################
+#     EC2 Deployment     #
+##########################
+
+# Create load balancer for Concourse web nodes
+web_lb = lb.LoadBalancer(
+    "concourse-web-load-balancer",
+    name=f"concourse-web-{stack_info.env_suffix}",
+    ip_address_type="dualstack",
+    load_balancer_type="application",
+    enable_http2=True,
+    subnets=operations_vpc["subnet_ids"],
+    security_groups=[
+        operations_vpc["security_groups"]["web"],
+    ],
+    tags=aws_config.merged_tags({"Name": f"concourse-web-{stack_info.env_suffix}"}),
+)
+
+TARGET_GROUP_NAME_MAX_LENGTH = 32
+web_lb_target_group = lb.TargetGroup(
+    "concourse-web-alb-target-group",
+    vpc_id=ops_vpc_id,
+    target_type="instance",
+    port=DEFAULT_HTTPS_PORT,
+    protocol="HTTPS",
+    health_check=lb.TargetGroupHealthCheckArgs(
+        healthy_threshold=2,
+        interval=10,
+        path="/",
+        port=str(DEFAULT_HTTPS_PORT),
+        protocol="HTTPS",
+    ),
+    name=f"concourse-web-{stack_info.env_suffix}"[:TARGET_GROUP_NAME_MAX_LENGTH],
+    tags=aws_config.tags,
+)
+concourse_web_acm_cert = acm.get_certificate(
+    domain="*.odl.mit.edu", most_recent=True, statuses=["ISSUED"]
+)
+concourse_web_alb_listener = lb.Listener(
+    "concourse-web-alb-listener",
+    certificate_arn=concourse_web_acm_cert.arn,
+    load_balancer_arn=web_lb.arn,
+    port=DEFAULT_HTTPS_PORT,
+    protocol="HTTPS",
+    default_actions=[
+        lb.ListenerDefaultActionArgs(
+            type="forward",
+            target_group_arn=web_lb_target_group.arn,
+        )
+    ],
+)
+
+# Create auto scale group and launch configs for Concourse web and worker
+web_launch_config = ec2.LaunchTemplate(
+    "concourse-web-launch-template",
+    name_prefix=f"concourse-web-{stack_info.env_suffix}-",
+    description="Launch template for deploying Concourse web nodes",
+    iam_instance_profile=ec2.LaunchTemplateIamInstanceProfileArgs(
+        arn=concourse_instance_profile.arn,
+    ),
+    image_id=concourse_web_ami.id,
+    vpc_security_group_ids=[
+        concourse_web_security_group.id,
+        operations_vpc["security_groups"]["web"],
+        consul_security_groups["consul_agent"],
+    ],
+    instance_type=InstanceTypes.medium,
+    key_name="salt-production",
```

Comment:
Should we use the stack_info for this?

Reply:
Possibly but in general we won't be sshing so it's not that important